include/cista/serialization.h

```diff
@@ -824,9 +841,30 @@ void recurse(Ctx&, basic_vector<T, Ptr, Indexed, TemplateSizeType>* el,
 // --- STRING ---
 template <typename Ctx, typename Ptr>
 void convert_endian_and_ptr(Ctx const& c, generic_string<Ptr>* el) {
+  using Type = generic_string<Ptr>;
+  auto str_convert_endian = [](Ctx const& ctx, typename Type::CharT* str,
+                               std::size_t const size) -> void {
+    if constexpr (sizeof(typename Type::CharT) > 1) {
+      for (std::size_t i = 0; i < size; ++i) {
+        ctx.convert_endian(str[i]);
+      }
+    }
+  };
+
   if (*reinterpret_cast<std::uint8_t const*>(&el->s_.is_short_) == 0U) {
     deserialize(c, &el->h_.ptr_);
     c.convert_endian(el->h_.size_);
+    try {
+      if constexpr (is_mode_disabled(Ctx::MODE, mode::UNCHECKED)) {
+        c.check_ptr(el->h_.ptr_,
+                    el->h_.size_ * sizeof(typename generic_string<Ptr>::CharT));
+      }
+      str_convert_endian(c, el->data(), static_cast<std::size_t>(el->h_.size_));
+    } catch (...) {
```

Comment:
why is the exception swallowed here?

Reply:
Checks are supposed to be made in check_state function, so I think it's expected convert_endian_and_ptr doesn't throw exceptions related to incorrect data.
Shouldn't it be swallowed?

Reply:
The idea of `convert_endian_and_ptr` is that it doesn't do any checking at all. If you want to do the endian conversion of the data inside the string after the check, then this would be a recursive step iterating the elements in the string.

Reply:
Alright. Then I'll rewrite this part